### DISCUSSION TITLE
Problem with ApplicationConfiguration source generator

### DISCUSSION THREAD

**Discussion by hamarb123** [dotnet/winforms#5927] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**

Yes, I was wanting to have code like `if (!Application.SetHighDpiMode(HighDpiMode.PerMonitorV2)) Application.SetHighDpiMode(HighDpiMode.PerMonitor);` in my Program file, but it seems like `ApplicationConfiguration.Initialize();` forces me to use its implementation.

**Describe the solution you'd like and alternatives you've considered**

Something like `<ApplicationHighDpiMode></ApplicationHighDpiMode>` or `<ApplicationHighDpiMode>unset</ApplicationHighDpiMode>` could disable that line in the source generator.

An alternative could be something like `<ApplicationHighDpiMode>PerMonitorV2OrPerMonitor</ApplicationHighDpiMode>` to specify the fallback I wrote above (perhaps including `SystemAware` as well).

The user could also just put it before `ApplicationConfiguration.Initialize();` and it would fail to set in there, but setting it twice seems like a bad solution to me since it's unnecessary.

**Will this feature affect UI controls?**

Not that I'm aware of.

**Comment by RussKie** [comment, marked as answer]
You can remove `ApplicationConfiguration.Initialize();` completely and write the code you would previously write.

**Reply by hamarb123** [reply]
That makes sense.
I'll do it with `_ = Application.SetHighDpiMode(HighDpiMode.PerMonitorV2) || Application.SetHighDpiMode(HighDpiMode.PerMonitor) || Application.SetHighDpiMode(HighDpiMode.SystemAware);` at the top of my Program.cs file so that way I can use it for the other features still.
